app/src/main/java/com/example/myapplication/MainActivity.java

```diff
@@ -46,14 +84,13 @@ protected void onCreate(Bundle savedInstanceState) {
     }
 
     void startLibraryActivity() {
-        LibraryService libraryService = new LibraryService();
-        libraryService.startMockChallenge(this, new CallbackService() {
-            @Override
-            public void onValidated(Context currentContext, String result) {
-                Intent intent = new Intent(currentContext, ResultActivity.class);
-                intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-                currentContext.startActivity(intent);
-            }
-        });
+        // clear result text
+        TextView resultTextView = findViewById(R.id.resultTextView);
+        resultTextView.setText("");
+
+        Intent intent = new Intent(this, LibraryActivity.class);
+        // Ref: https://stackoverflow.com/a/48177487
+        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        activityLauncher.launch(intent);
```

Comment:
Use the `activityLauncher` to launch a `LibraryActivity`.

Note the omission of `Intent.FLAG_ACTIVITY_NEW_TASK` when specifying a launch `Intent`. Android [documentation](https://developer.android.com/reference/android/content/Intent#FLAG_ACTIVITY_NEW_TASK) states that "this flag can not be used when the caller is requesting a result from the activity being launched."

Reply:
In your example, The Challenge Activity is being started from the App, in practice the app will not have access to start CardinalMobileSDK's Challenge Activity from the app's activity -- reason being: before starting the Activity, the SDK validates the cRes and couple other things. 

So if we can proof out the possibility of Activity Result from UIInteractionFactory.java in 'testLibrary' module, then we can test it out with the CardinalMobileSDK.

Reply:
Understood! There may be a way for the Cardinal SDK to encapsulate the launch of `LibraryActivity` e.g. by providing a `CardinalLauncher` that leverages the Activity Result API internally.

Reply:
So I tested with passing a reference of apps's ActivityResultLauncher to the SDK and start activity from that (since ActivityResultLauncher it can only be implemented in an activity)-- it works but when the process kill happens in background, it falls back to the first activity (app). Which is not the best case BUT better than crash. 

I will test more scenarios and do further research on implications on this and hopefully this can be a possible solution.

Reply:
👍 . I spiked a sample solution in another PR with this branch as the merge base: [Use Activity Result API with Encapsulation – #2](https://github.com/sshropshire/process-kill-example/pull/2). There are ways to refine it further, but it should be possible to encapsulate the Activity result API.

There's also an example solution [here in the docs](https://developer.android.com/training/basics/intents/result#separate) that may be the best route. It's less encapsulated and may be more developer friendly in the long run.